pkg/openstack/nova.go

```diff
@@ -80,6 +80,25 @@ func ReconcileNova(ctx context.Context, instance *corev1beta1.OpenStackControlPl
 		instance.Spec.Nova.Template.TopologyRef = instance.Spec.TopologyRef
 	}
 
+	// When there's no NotificationsBusInstance referenced in the Service Template,
+	// inject the top-level one. Allow overriding notifications bus instance locally
+	// in the services templates, unless it is "" top scope, which always wins.
```

Comment:
This feels unnecessarily complicated. I suggest to do it simply:
* if the value in the template is not `nil` the use that value (either `""` or `"<name>"`)
* if the value in the template is `nil` then copy the value from the top level field to the template field (again that can be either `""` or `"<name>"`.

Reply:
this implements the designed CRD, let's move discussions there